query-engine/query-engine-node-api/src/engine.rs

```diff
@@ -416,17 +413,25 @@ impl QueryEngine {
     /// Loads the query schema. Only available when connected.
     #[napi]
     pub async fn sdl_schema(&self) -> napi::Result<String> {
+        let dispatcher = self.logger.dispatcher();
+        let recorder = self.logger.recorder();
+
         async_panic_to_js_error(async move {
             let inner = self.inner.read().await;
             let engine = inner.as_engine()?;
 
             Ok(render_graphql_schema(engine.query_schema()))
         })
+        .with_subscriber(dispatcher)
+        .with_optional_recorder(recorder)
         .await
     }
```

Comment:
Unrelated, but we can probably get rid of this method, unless certain third-party clients depend on it.